robolectric/src/main/java/org/robolectric/internal/bytecode/ShadowWrangler.java

```diff
@@ -44,12 +44,13 @@ public Object run(Object instance, Object roboData, Object[] params) throws Exce
   private static final MethodHandle NO_SHADOW_HANDLE = constant(Object.class, NO_SHADOW);
   private final ShadowMap shadowMap;
   private final Map<Class, MetaShadow> metaShadowMap = new HashMap<>();
-  private final Map<String, Plan> planCache = new LinkedHashMap<String, Plan>() {
-    @Override
-    protected boolean removeEldestEntry(Map.Entry<String, Plan> eldest) {
-      return size() > 500;
-    }
-  };
+  private final Map<String, Plan> planCache =
+      Collections.synchronizedMap(new LinkedHashMap<String, Plan>() {
```

Comment:
Won't this create contention? Not sure if this will significantly affect performance though. If it will, we might want to look for a more appropriate concurrent LRU cache implementation.